subsys/bluetooth/mesh/settings.c

```diff
@@ -1233,6 +1303,97 @@ static void store_pending_keys(void)
 	}
 }
 
+#if defined(CONFIG_BT_MESH_PROVISIONER)
```

Comment:
One problem is seems to be that the static `node_updates` array is using `CONFIG_BT_MESH_NODE_COUNT` which is only defined when `CONFIG_BT_MESH_PROVISIONER` is defined. Any way around this?

Reply:
You could have something like:
```
#if defined(CONFIG_BT_MESH_NODE_COUNT)
#define NODE_COUNT CONFIG_BT_MESH_NODE_COUNT
#else
#define NODE_COUNT 0
#endif
```

And then use NODE_COUNT instead of CONFIG_BT_MESH_NODE_COUNT

Reply:
@jhedberg One way I guess would be to still have a `node_updates[1]` if `CONFIG_BT_MESH_PROVISIONER` is not set. I think this would be optimized away but it feels like sort of strange to do it like that.